pkg/bootstrap/platform/gcp.go

```diff
@@ -286,7 +288,7 @@ func zoneToRegion(z string) (string, error) {
 // Locality returns the GCP-specific region and zone.
 func (e *gcpEnv) Locality() *core.Locality {
 	var l core.Locality
-	if e.shouldFillMetadata {
+	if fillMetadata {
```

Comment:
I think this may be broken (before and after this change). If Metadata isn't called first then `fillMetadata` is always empty.

It may be worth just switching to `pkg/lazy` which makes these errors ~impossible

Reply:
sounds good, lets fix it in a follow up commit